### PR TITLE
prevent direct navigation to contribution thank you page

### PR DESCRIPTION
**Pull request by michaelbjacobson**
## What does this PR do?

Per [this ticket](https://trello.com/c/LBpPmGmy), this PR adds some additional conditional logic to the React Router components on the contributions landing page. If a user points their browser directly to `https://support.theguardian.com/uk/thankyou`, they'll be redirected to `https://support.theguardian.com/uk/contribute`. This doesn't affect the post-contribution redirect to the thank you page.

_n.b. for testing purposes etc. we can directly navigate to the thank you page by appending the_ `no-redirect` _param to the query string, eg._ `https://support.theguardian.com/uk/thankyou?no-redirect`.

### CODE REVIEW

**Comment by michaelbjacobson** [comment]
[Make it **impossible** to reach the thank you page without making a contribution](https://trello.com/c/LBpPmGmy/1542-make-it-impossible-to-reach-the-thank-you-page-without-making-a-contribution)

**Review by TomPretty — APPROVED** [review]
Nice! Looks great! 👍

File: support-frontend/assets/pages/contributions-landing/components/ContributionFormContainer.jsx [hunk 1]
```diff
@@ -96,6 +97,7 @@ function withProps(props: PropTypes) {
     // This is because going 'back' to the /contribute page is not helpful, and the client-side routing would redirect
     // back to /thankyou given the current state of the redux store.
     // The effect is that clicking back in the browser will take the user to the page before they arrived at /contribute
+    storage.setSession('isPaymentComplete', 'true');
```
Comment (tomrf1) [inline support-frontend/assets/pages/contributions-landing/components/ContributionFormContainer.jsx:100]: unfortunately this breaks paypal (one off), which requires a redirect. I'm trying to understand how we do preserve some session storage when this happens...

Comment (michaelbjacobson) [reply support-frontend/assets/pages/contributions-landing/components/ContributionFormContainer.jsx:100]: Dang. I only tested it using the mock Stripe details locally. Do you want to pair on this in about half an hour? No worries if not!

Comment (tomrf1) [reply support-frontend/assets/pages/contributions-landing/components/ContributionFormContainer.jsx:100]: e.g. this session storage does persist through to the thankyou page: https://github.com/guardian/support-frontend/blob/724acf30ad4dd1c0291dd8222e32cd35aba07cb3/support-frontend/assets/pages/contributions-landing/contributionsLandingActions.js#L283
I'm not sure what difference is here

Comment (TomPretty) [reply support-frontend/assets/pages/contributions-landing/components/ContributionFormContainer.jsx:100]: Ahhhh....isn't there a function that runs after the paypal redirect?

Comment (TomPretty) [reply support-frontend/assets/pages/contributions-landing/components/ContributionFormContainer.jsx:100]: Is it too hacky to do something here? https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/pages/contributions-landing/contributionsLandingInit.js#L179

Comment (TomPretty) [reply support-frontend/assets/pages/contributions-landing/components/ContributionFormContainer.jsx:100]:
```
dispatch(setGuestAccountCreationToken(window.guardian.guestAccountCreationToken));
dispatch(setThankYouPageStage('thankYouSetPassword'));\
storage.setSession('isPaymentComplete', 'true');
```

Comment (tomrf1) [reply support-frontend/assets/pages/contributions-landing/components/ContributionFormContainer.jsx:100]: `guestAccountCreationToken` only applies to a subset of our contributors

Comment (TomPretty) [reply support-frontend/assets/pages/contributions-landing/components/ContributionFormContainer.jsx:100]: ahh of course...hmmmm...

**Review by tomrf1 — APPROVED** [review]
🚀

**Pull request merged**